doc/research/nextcloud_integration.md

```diff
@@ -9,45 +11,208 @@
 
-```bash
-help
+Now the circle has to be added to the shares for the map directory.
+This can be done with the share API which is documented [here](https://docs.nextcloud.com/server/latest/developer_manual/client_apis/OCS/ocs-share-api.html).
+Following API call can be used:
 ```
+"method": "post",
+"headers": [
+  "ocs-apirequest": true
+],
+"body": {
+  "path": "PermaplanT/Maps/map_01",
+  "shareType": 7,
+  "shareWith": "<circleId>"
+},
+"scheme":	"https",
+"host": "cloud.permaplant.net",
+"filename": "/ocs/v2.php/apps/files_sharing/api/v1"
+```
+
+### Components
+
+There are a number of components which help with interacting with Nextcloud files.
+
+#### Image Components
+
+There are four components that handle fetching and displaying images from Nextcloud.
+
+- NextcloudImage: This component fetches an image from the user directory in Nextcloud and renders it.
+- PublicNextcloudImage: This component fetches and renders an image which is available on a public share.
+- NextcloudKonvaImage: This component fetches an image from the user directory in Nextcloud and creates a shape for Konva (Canvas library used in PermplanT).
+- PublicNextcloudKonvaImage: This component fetches an image from the Nextcloud public share and creates a shape for Konva.
+
+#### File Selection
+
+Component which shows the content of a Nextcloud directory and lets the user choose a file.
+
+Will be implemented in #475.
+
+### Further Related Resources
+
+- https://docs.nextcloud.com/server/latest/developer_manual/client_apis/WebDAV/comments.html
+- https://docs.oracle.com/cd/E12839_01/portal.1111/e10235/webdav007.htm#POUSR1607
+- https://docs.nextcloud.com/server/25/user_manual/en/files/access_webdav.html#
+- http://www.webdav.org/specs/rfc4918.html
+
+## Limitations
+
+### CORS (Cross-Origin Resource Sharing)
+> "CORS is an HTTP-header based mechanism that allows a server to indicate any origins (domain, scheme, or port) other than its own from which a browser should permit loading resources."
+-- [Mozilla MDN](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS)
+
+Nextcloud offers no options to change the CORS configuration. This means that all requests coming from a different origin than the one Nextcloud is hosted on are blocked.
 
-## calendar
+The PermaplanT production environment has following structure:
+- Frontend: "https://www.permaplant.net".
+- Backend: "https://www.permaplant.net/api"
+- Nextcloud: "https://cloud.permaplant.net"
+- Keycloak: "https://auth.permaplant.net/realms/PermaplanT"
+
+This means the origin for Nextcloud differs from the origin of the PermaplanT frontend. Consequently CORS has to be configured to allow requests from the frontend origin otherwise the requests to Nextcloud are blocked by the browser.
```

Comment:
```suggestion
This means the origin for Nextcloud differs from the origin of the PermaplanT frontend.
Consequently CORS has to be configured to allow requests from the frontend origin otherwise the requests to Nextcloud are blocked by the browser.
```